modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/SearchLocation/usePlaces.es.js

```diff
@@ -15,8 +15,23 @@
 import {useEffect, useRef, useState} from 'react';
 
 const loadScript = (readOnly, elementId, googlePlacesAPIKey, callback) => {
-	const script = document.createElement('script');
-	script.setAttribute('type', 'text/javascript');
+	let script = document.getElementById('googleMapsScript');
+
+	if (!script) {
+		script = document.createElement('script');
```

Comment:
It seems the created element is not being removed from the document after the component is unmounted

Reply:
I believe that's intentional, @miltonmc. If we removed the script, wouldn't we have to download Google's files every time we mount back the component? I believe this way, we keep the files in memory for as long as the page is being rendered. Plus, this way we could have more than one Search Location fields in the form, but we just make use of one script. Does that make sense to you?

Reply:
My concern is related do a possible memory leaks and/or performance degradation by adding things to the document that may not be in use. I believe that once it was downloaded by the browser this file won't be downloaded again due to its [cache policy](https://pressidium.com/blog/2017/browser-cache-work/). What do you think?

Reply:
You're right about the browser's cache. Thanks for reminding me. And, regarding adding the script even when it's not being used, do you mean during pagination or when the field is hidden?

Reply:
Actually I meant once it was added to the document and not removed, we could navigate to other areas of the app with no relations to that (like the system settings) and the content would still be present into the document.

Reply:
Geez... you're right! That's awful. I just checked and the script's still there wherever I go, until I fire a hard refresh on the page I'm at. I thought, once I went to another JSP, the entire page would be replaced. Is this a browser, javascript thing? What's the deal?

Reply:
I'm not 100% sure about that, but I think it's related to the JSP and the way a portlet behaves.

As I understand, portlet is kind of a container for a JSP that has its content replaced every time a JSP load things in that portlet.

So if we have a page like:
```html
<html>
  <body>
    <div id="my-portlet">
      hello world!
    </div>
  </body>
</html>
```

Supposed that my portlet points to the `div` with the `id="my-portlet"`, so it would be able to replace its content without the need of reload the entire document (a concept similar to the one used by SPAs - Single Page Applications). 

That's why the elements attached to the `document` won't disappear.

Reply:
I see. That's cumbersome. ☹️

Reply:
Forgot to ask. Do you need me to change this right now or can we open a bug to address this behavior, since that's not an impedibug?

Reply:
All that you need to fix this behaviour is call the `remove()` into the function returned by the `useEffect` on `usePlaces.es.js`:
```javascript
const usePlaces = ({elementId, googlePlacesAPIKey, isReadOnly, onChange}) => {
	...

	useEffect(() => {
		...

		return () => {
			...
			document.getElementById('googleMapsScript').remove()
		};
	}, []);
};
```

Reply:
Sure. I'll make these changes right away.

Reply:
@miltonmc, I just applied your suggestion and the scripts are still there.
<img width="1344" alt="Screen Shot 2021-08-10 at 10 11 36" src="https://user-images.githubusercontent.com/11450348/128872848-276cb00d-7851-40da-821f-5f04bd7fc267.png">
Am I missing something?

Reply:
Sorry I've forgot to mention that is better verifies element existence before you remove it, because if you have more than one of this component rendered, the component  may try to remove an already remover element and it will log an exception into browser's console (`Uncaught TypeError: Cannot read property 'remove' of null`)

I think you will be fine if you do:
```javascript
document.getElementById('googleMapsScript')?.remove()
```

Reply:
That's actually what I did while testing, but still the scripts weren't removed.